sqlflow/client.py

```diff
@@ -114,54 +128,59 @@ def execute(self, operation):
         except EnvExpanderError as e:
             raise e
 
-    @classmethod
-    def display(cls, stream_response):
+    
+    def read_fetch_response(self, job_id):
+        req = pb.FetchRequest()
+        pb_job = pb.Job(id=job_id)
+        pb_job.id = job_id
+        req.job.CopyFrom(pb_job)
+        while True:
+            response = self._stub.Fetch(req)
+            for log in response.logs.content:
+                _LOGGER.info(log)
+            if response.eof:
+                break
+            req = response.updated_fetch_since
+
+    def display(self, stream_response):
         """Display stream response like log or table.row"""
         compound_message = CompoundMessage()
         while True:
             try:
                 first = next(stream_response)
             except StopIteration:
                 break
-            if first.WhichOneof('response') == 'message':
-                # if the first line is html tag like,
-                # merge all return strings then render the html on notebook
+            oneof_first = first.WhichOneof('response')
+            reader = StreamReader(stream_response)
+            if oneof_first == 'message':
                 if re.match(r'<[a-z][\s\S]*>.*', first.message.message):
                     resp_list = [first.message.message]
-                    for res in stream_response:
-                        if res.WhichOneof('response') == 'eoe':
-                            _LOGGER.info("end execute %s, spent: %d" % (res.eoe.sql, res.eoe.spent_time_seconds))
-                            compound_message.add_html('\n'.join(resp_list), res)
-                            break
+                    for res in reader.read():
                         resp_list.append(res.message.message)
                     from IPython.core.display import display, HTML
                     display(HTML('\n'.join(resp_list)))
                 else:
-                    all_messages = []
-                    all_messages.append(first.message.message)
-                    eoe = None
-                    for res in stream_response:
-                        if res.WhichOneof('response') == 'eoe':
-                            _LOGGER.info("end execute %s, spent: %d" % (res.eoe.sql, res.eoe.spent_time_seconds))
-                            eoe = res
-                            break
-                        _LOGGER.debug(res.message.message)
-                        all_messages.append(res.message.message)
-                    compound_message.add_message('\n'.join(all_messages), eoe)
+                    _LOGGER.info(first.message.message)
+                    for res in reader.read():
+                        _LOGGER.info(res.message.message)
+            elif oneof_first == 'job':
+                self.read_fetch_response(first.job.id)
```

Comment:
Does this line duplicate with line 178?

Reply:
No, the first response iteration may be `message` type which contains job logs, and the last response is `job` type.